### PR TITLE
[Merged by Bors] - feat: port Data.List.FinRange

### PR DESCRIPTION
**Pull request by mo271**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/List/FinRange.lean [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    rw [← nthLe_map_rev]
+    congr
+    · rw [nthLe_finRange]
+      rfl
+    · rw [length_finRange]
+      exact h
+#align list.map_nth_le List.map_nth_le
+-/
+
+theorem of_fn_eq_pmap {α n} {f : Fin n → α} :
```
Comment (jcommelin) [inline Mathlib/Data/List/FinRange.lean:55]: Also elsewhere
```suggestion
theorem ofFn_eq_pmap {α n} {f : Fin n → α} :
```

File: Mathlib/Data/List/FinRange.lean [hunk 2]
```diff
@@ -25,48 +25,46 @@ namespace List
...
-
-@[simp]
+  rw [map_cons, map_coe_finRange, range_succ_eq_map, Fin.val_zero, ← map_coe_finRange, map_map,
+    map_map]
+  simp only [Function.comp, Fin.val_succ]
+#align list.fin_range_succ_eq_map List.finRange_succ_eq_map
+
+-- Porting note TODO: we have List.finRange_map_get in Data.List.Range, which seems to be exactly
+-- this but currently not a simp lemma. Should we remove map_nth_le, make finRange_map_get
+-- a simp lemma and align map_nth_le with finRange_map_get?
```
Comment (jcommelin) [inline Mathlib/Data/List/FinRange.lean:42]: Yes, `nthLe` is deprecated, so we always align with `get`-lemmas instead.
If you add the `#align`-statement in the other file, please leave a porting note here with a pointer.

Comment (mo271) [reply Mathlib/Data/List/FinRange.lean]: done, making it at `simp` lemma like it used to be

File: Mathlib/Data/List/FinRange.lean (hunk 1 above)
Comment (mo271) [reply Mathlib/Data/List/FinRange.lean]: done in all places, and also `of_injective` to `ofInjective`

File: Mathlib/Data/List/FinRange.lean [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+-- Porting note : `map_nth_le` moved to `List.finRange_map_get` in Data.List.Range
+
+theorem ofFn_eq_pmap {α n} {f : Fin n → α} :
+    ofFn f = pmap (fun i hi => f ⟨i, hi⟩) (range n) fun _ => mem_range.1 := by
+  (rw [pmap_eq_map_attach];
+    exact ext_get (by simp) fun i hi1 hi2 => by
+        simp [get_ofFn f ⟨i, hi1⟩])
+#align list.of_fn_eq_pmap List.ofFn_eq_pmap
+
+theorem of_fn_id (n) : ofFn id = finRange n :=
```
Comment (jcommelin) [inline Mathlib/Data/List/FinRange.lean:49]:
```suggestion
theorem ofFn_id (n) : ofFn id = finRange n :=
```

**Review by jcommelin — COMMENTED** [review]
One more naming issue. After that, LGTM

bors d+

File: Mathlib/Data/List/FinRange.lean (hunk 3 above)
Comment (mo271) [reply Mathlib/Data/List/FinRange.lean]: done

**Comment by mo271** [comment]
Thanks, bors r+

**Pull request closed without merging**